examples/runner-public/key.tf

```diff
@@ -1,25 +1,38 @@
-resource "tls_private_key" "ssh" {
-  algorithm = "RSA"
-}
+####################################
+# Key Pair
+####################################
 
-resource "local_file" "public_ssh_key" {
-  depends_on = ["tls_private_key.ssh"]
+locals {
+  key_pair_specified      = "${var.key_pair_name == ""}"
+  key_pair_write          = "${local.key_pair_specified && var.write_private_key}"
+  generated_key_pair_name = "${var.environment}-gitlab-runner"
+  key_pair_name           = "${ local.key_pair_specified ? local.generated_key_pair_name : var.key_pair_name }"
+  key_pair_path           = "${path.module}/generated}"
```

Comment:
Last `}` should not be there

Reply:
rebased against your other fixes so this should be good now